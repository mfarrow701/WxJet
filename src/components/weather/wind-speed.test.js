import React from 'react';
import {shallow} from 'enzyme';
import WindSpeed from './wind-speed';

describe('Wind speed component', () => {
    let component;

    const renderWrapper = () => {
        component = shallow(<WindSpeed/>);
    };

    beforeEach(() => {
        renderWrapper();
    });

    it('should render the wind speed component after load', () => {
        expect(component).toMatchSnapshot();
        expect(component.find('.Wind-Speed').text()).toEqual('--kts');
    });

    it('should render the wind speed component with a custom value', () => {
        component.setProps({value: 7});
        expect(component.find('.Wind-Speed').text()).toEqual('7m/s');
    });

    it('should toggle between knots and metres per second on click', () => {
        component.setProps({value: 7});
        component.find('.Wind-Speed').simulate('click');
        expect(component.find('.Wind-Speed').text()).toEqual('14kts');
        component.find('.Wind-Speed').simulate('click');
        expect(component.find('.Wind-Speed').text()).toEqual('7m/s');
    });
});
