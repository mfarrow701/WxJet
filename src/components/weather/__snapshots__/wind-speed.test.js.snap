// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wind speed component should render the wind speed component after load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WindSpeed />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "--kts",
      "className": "Wind-Speed",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "--kts",
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "--kts",
        "className": "Wind-Speed",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "--kts",
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
